dashboard/app/entities_spanner.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"google.golang.org/api/iterator"
+)
+
+// This file contains definitions of entities stored in spanner.
+
+type CoverageHistory struct {
+	instrumented map[string]int64
+	covered      map[string]int64
+}
+
+// MergedCoverage uses dates, not time.
+func MergedCoverage(ctx context.Context, ns string, fromDate, toDate time.Time) (*CoverageHistory, error) {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	client, err := spanner.NewClient(ctx, "projects/"+projectID+"/instances/syzbot/databases/coverage")
+	if err != nil {
+		panic(fmt.Sprintf("spanner.NewClient() failed: %s", err.Error()))
+	}
+	defer client.Close()
+
+	stmt := spanner.Statement{
+		SQL: `select
+			DATE_TRUNC('day', dateto) as targetdate,
+			cast(sum(instrumented) as INTEGER) as instrumented,
+			cast(sum(covered) as INTEGER) as covered
+		from "` + ns + `_files"
+	where DATE_TRUNC('day', dateto)>DATE_TRUNC('day', $1) and DATE_TRUNC('day', dateto)<DATE_TRUNC('day', $2)
```

Comment:
Nit: do we need to truncate dates here? Will `dateto >= $1` not work?

Reply:
Why not use https://pkg.go.dev/time#Time.Format or directly ask https://pkg.go.dev/time#Time.Year etc?

Reply:
I experimented with time.Format() a few days ago and the error was something like "can't compare string and date". (some time passed thus I'm 95% sure)
civil.Date() was recommended by examples.

Done.